analyzers/tests/SonarAnalyzer.UnitTest/Rules/ExpressionComplexityTest.cs

```diff
@@ -34,6 +34,66 @@ public class ExpressionComplexityTest
                 .WithOptions(ParseOptionsHelper.FromCSharp8)
                 .Verify();
 
+        [DataTestMethod]
+        [DataRow("==")]
+        [DataRow("!=")]
+        [DataRow("<")]
+        [DataRow("<=")]
+        [DataRow(">")]
+        [DataRow(">=")]
+        public void ExpressionComplexity_TransparentComparissionOperators(string @operator) =>
+            builderCS.AddSnippet($$$"""
+            class C
+            {
+                public void M()
+                {
+                    var x = true && true && (1 {{{@operator}}} (true ? 1 : 1));         // Compliant (Make sure, the @operator is not increasing complexity)
+                    var y = true && true && true && (1 {{{@operator}}} (true ? 1 : 1)); // Noncompliant {{Reduce the number of conditional operators (4) used in the expression (maximum allowed 3).}}
+                }
+            }
+            """)
+            .Verify();
+
+        [DataTestMethod]
+        [DataRow("o", "??")]
+        [DataRow("i", "|")]
+        [DataRow("i", "^")]
+        [DataRow("i", "&")]
+        [DataRow("i", ">>")]
+        [DataRow("i", ">>>")]
+        [DataRow("i", "<<")]
+        [DataRow("i", "+")]
+        [DataRow("i", "-")]
+        [DataRow("i", "*")]
+        [DataRow("i", "/")]
+        [DataRow("i", "%")]
+        public void ExpressionComplexity_TransparentBinaryOperators(string parameter, string @operator) =>
+            builderCS.AddSnippet($$"""
+            class C
+            {
+                public void M(int i, object o)
+                {
+                    var x = true && true && (({{parameter}} {{@operator}} (true ? {{parameter}} : {{parameter}})) == {{parameter}});         // Compliant
+                    var y = true && true && true && (({{parameter}} {{@operator}} (true ? {{parameter}} : {{parameter}})) == {{parameter}}); // Noncompliant
+                }
+            }
+            """)
+            .WithOptions(ParseOptionsHelper.FromCSharp11)
+            .Verify();
+
+        [TestMethod]
+        public void ExpressionComplexity_RootBinaryOperators() =>
+            builderCS.AddSnippet("""
+            class C
+            {
+                public void M(object o)
+                {
+                    var x = true && true && true && (((true ? o : o) as bool?) ?? true); // Compliant. The ? : expression is inside the binary as expression. The as expression starts a new root.
```

Comment:
I think this one should go directly to the TestCase file too. Why do we need a separate method?